### PR TITLE
Issue #3476: Remove the search block default title. Update old blocks…

### LINKED ISSUE #3476: Remove search block default title
In https://github.com/backdrop/backdrop-issues/issues/2923 we discussed how there are a lot of instances of the word "Search". Ideally, the search form block would have no title. We can remove it, but we'd also need an update hook to update all existing layouts to add a custom block title for any block that had previously used the default. This could be a bit tricky, so in the hopes of getting https://github.com/backdrop/backdrop-issues/issues/2923 in for Backdrop 1.12, I've split this off into a separate issue. 

The problem: 
screen shot 2019-01-12 at 11 08 14 am

The solution:
screen shot 2019-01-12 at 11 09 14 am

Steps to test:
1. Fresh install of backdrop (or, untouched sandbox)
2. Confirm there is no search block on any layout
3. Add a search block -- ignore all settings on the configure block form, just add it blindly.
4. Confirm there is no title shown for the search block.

---
PR: https://github.com/backdrop/backdrop/pull/2470

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/3476

### CODE REVIEW

File: core/modules/search/search.install [hunk 1]
```diff
@@ -201,6 +201,27 @@ function search_update_1001() {
...
+ * Set all search blocks with 'default' title to 'custom' title with 'Search'.
+ */
+function search_update_1002() {
+  $config_names = config_get_names_with_prefix('layout.layout.');
+  foreach ($config_names as $config_file) {
+    $config = config($config_file);
+    $data = $config->get();
+    foreach ($data['content'] as $hash => $block) {
+      if ($block['plugin'] == 'search:form'
+          && $block['data']['settings']['title_display'] == 'default') {
```
Comment (klonos) [inline core/modules/search/search.install]: Not sure if such splitting of expressions within conditionals is according to our coding standards.

Comment (jenlampton) [reply core/modules/search/search.install]: I'm not sure either, but I thought I'd send it by @quicksketch and see what he thought. I do this all the time in my custom code (so I can read it clearly) and did it here by force-of-habit. After I looked at it I thought, eh, maybe? and subimtted the PR anyway :)

Comment (quicksketch) [reply core/modules/search/search.install]: While less preferable than using separate variables, I found that we have at least 45 instances of doing this in code currently. I think updating the coding standards is in order to include reasonable separation of conditions on different lines by `&&` or `||` operators.

Comment (jenlampton) [reply core/modules/search/search.install]: If i wanted to use separate variables, would it need to add 4 extra lines, like this?
```php
      $block_title_default = FALSE;
      if ($block['data']['settings']['title_display'] == 'default') {
        $block_title_default = TRUE;
      }
      if ($block['plugin'] == 'search:form' && $block_title_default) {
```

Or could we do use a ternary even though it's more than 80 characters?
```php
      $default = $block['data']['settings']['title_display'] == 'default'? TRUE: FALSE;
      if ($block['plugin'] == 'search:form' && $default) {
```

File: core/modules/search/search.install [hunk 2]
```diff
@@ -201,6 +201,27 @@ function search_update_1001() {
...
+ * Set all search blocks with 'default' title to 'custom' title with 'Search'.
+ */
+function search_update_1002() {
+  $config_names = config_get_names_with_prefix('layout.layout.');
+  foreach ($config_names as $config_file) {
+    $config = config($config_file);
+    $data = $config->get();
+    foreach ($data['content'] as $hash => $block) {
+      $default = $block['data']['settings']['title_display'] == 'default'? TRUE: FALSE;
+      if ($block['plugin'] == 'search:form' && $default) {
```
Comment (docwilmot) [inline core/modules/search/search.install]: This is failing on upgrade tests only. Presumably the upgrade test may create blocks without settings and thus no `'settings'` key. This should work better:
```
    foreach ($data['content'] as $hash => $block) {
      if ($block['plugin'] == 'search:form' && isset($block['data']['settings']['title_display']) && $block['data']['settings']['title_display'] == 'default') {
         ...
```
Or maybe split into two lines:
```
    foreach ($data['content'] as $hash => $block) {
      if ($block['plugin'] == 'search:form') {
        if (isset($block['data']['settings']['title_display']) && $block['data']['settings']['title_display'] == 'default') {
         ...
```

Comment (jenlampton) [reply core/modules/search/search.install]: I ended up going with 3 lines to keep the character limit as close to 80 as seemed reasonable: 
```php
      if ($block['plugin'] == 'search:form') {
        $has_title_display = isset($block['data']['settings']['title_display']);
        if ($has_title_display && $block['data']['settings']['title_display'] == 'default') {
```

File: core/modules/search/search.install (hunk 1 above)
Comment (ghost) [reply core/modules/search/search.install]: One option is:

```php
$block_title_display = $block['data']['settings']['title_display'];
if ($block['plugin'] == 'search:form' && $block_title_display == 'default') {
```

However I tend to steer away from using a variable like this when it's just used in one place only. Not sure if that's right or wrong though...

> Or could we do use a ternary even though it's more than 80 characters?

Code can be longer than 80 characters, it's only comments that are affected by that restriction (to my knowledge).

**Comment by jenlampton** [comment]
closing for new sandbox.

**Comment by jenlampton** [comment]
sandbox please

**Comment by laryn** [comment]
Thanks @jenlampton, @quicksketch, @docwilmot, @BWPanda, and @klonos! Merged into 1.x and 1.19.x.

**Pull request merged**